src/hangar/metadata.py

```diff
@@ -72,7 +72,15 @@ def __len__(self):
         int
             number of metadata key/value pairs.
         '''
-        return len(self._Query.metadata_names())
+        if not self.__is_conman:
```

Comment:
Two things to note here. `self.__is_conman` was the old name we followed. I noticed you had changed it to `self._is_conman`.